src/cutadapt/kmer_heuristic.py

```diff
@@ -0,0 +1,198 @@
+import io
+import itertools
+import sys
+from typing import List, Optional, Set, Tuple
+from collections import defaultdict
+
+
+def kmer_possibilities(sequence: str, chunks: int) -> List[Set[str]]:
+    """
+    Partition a sequence in almost equal sized chunks. Return all possibilities.
+
+    Example sequence ABCDEFGH with 3 chunks. Possibilities:
```

Comment:
You could turn this into a doctest

Reply:
I did so, but this doctest turns out to be rather sensitive to the order within the sets and this is not determinate unfortunately, so the doctest will only succeed randomly.